qrenderdoc/Windows/Dialogs/VirtualFileDialog.cpp

```diff
@@ -590,9 +590,19 @@ VirtualFileDialog::VirtualFileDialog(ICaptureContext &ctx, QString initialDirect
 
   ui->buttonBox->button(QDialogButtonBox::Ok)->setDefault(false);
 
+  QModelIndex index;
+
+  if(!initialDirectory.isEmpty() && !ctx.Replay().CurrentRemote()->IsADB())
```

Comment:
As I mentioned before this handling should not check for Android specifically, it should apply this logic regardless of the type of remote host. The resulting error from an unrecognised path is fine because it's essentially user input and we have the fallback path below to select the home folder if an invalid index is returned.

Reply:
I used this check because in case of Android remote host we don't need to call the **indexForPath** method at all. If we call it, we will get that malformed path exception every time.
How should it work for Android?

Reply:
Yes, the problem is that the previous local path is still there when the user switches to Android and tries to open the file browser. I assumed to open it regardless of what executable path is written there. Would it be better solution if we clear these input fields and values when the remote host is changed?

Reply:
I also get this exception when I've already launched an Android application, but I want to open the file browser again to launch another one (because the package name is set as executable path and it does not contain '/' character). So it does not only happens when the remote host is switched.

Reply:
I'm unclear exactly what problem you're having. I've pushed a fix for the two bugs I've seen - that the VirtualFileDialog didn't fallback to the home directory, and that the CaptureDialog didn't properly trim the pathname out of a selected executable file for either windows paths or android paths (with no path at all).

Can you please re-test and give precise details about what problems you still encounter, if any.

Reply:
OK, that sounds like the bug that was fixed in 3893fc2567353a26551655cedd50eee64f920742.

Does that still happen if you delete the broken saved `lastCapturePath` in `%APPDATA%/qrenderdoc/UI.config` or `~/.local/share/qrenderdoc/UI.config` or wherever `UI.config` is stored on your system? The fix applies to the *saving* of that value, so it won't properly apply until you've either gone through the selection and saved a new correct `lastCapturePath`, or manually fixed the config.

Also just for future information, an exception is a different thing to an error message. It is a bit confusing to call an error message an exception (and in the original bug report I didn't understand what you meant because of this).

Reply:
After deleting the UI.config file it works fine now, thank you!